src/vulkan/Context.zig

```diff
@@ -72,13 +74,124 @@ pub fn init(
     };
 
     const instance_handle = try self.vkb.createInstance(&instance_create_info, null);
-    self.vki = try InstanceDispatch.load(instance_handle, self.vkb.dispatch.vkGetInstanceProcAddr);
-    self.instance = Instance.init(instance_handle, &self.vki);
+    const vki = try InstanceDispatch.load(instance_handle, self.vkb.dispatch.vkGetInstanceProcAddr);
+    self.instance = Instance.init(instance_handle, &vki);
+
+    var surface: vk.SurfaceKHR = undefined;
+    if (glfw.createWindowSurface(instance_handle, window, null, &surface) != 0)
+        return error.SurfaceLostKHR;
 
+    self.physical_device = try pickPhysicalDevice(self, surface, allocator);
```

Comment:
Can't you do `self.pickPhysicalDevice(surface, allocator)` instead here?
Alternatively, this should maybe just be a helper function, rather than a method, if it's only used by the class itself.

Also, allocator is usually the first parameter by convention.